docs/parameters.md

```diff
@@ -0,0 +1,322 @@
+The resulting structure represents an arbitrary JSON object, which is assumed to 
+be valid for a particular broker. 
+The Service Catalog does not enforce any extra limitations on the format and content 
+of this structure.
+
+## Design
+
+When you create an `Instance` or a `Binding`, you can set parameters to be passed 
+to the corresponding broker.
+To set parameters, include the `parameters` or `parametersFrom` field in the spec.
+
+### Basic example
+
+Let's say we want to create an `Instance` of EC2 running on AWS using a
+[corresponding broker](https://github.com/cloudfoundry-samples/go_service_broker) 
+which implements the Open Service Broker API.
+
+A typical provisioning request for this broker looks [like this](https://github.com/cloudfoundry-samples/go_service_broker/blob/master/bin/curl_broker.sh):
+```bash
+curl -X PUT http://username:password@localhost:8001/v2/service_instances/instance_guid-111 -d '{
+  "service_id":"service-guid-111",
+  "plan_id":"plan-guid",
+  "organization_guid": "org-guid",
+  "space_guid":"space-guid",
+  "parameters": {"ami_id":"ami-ecb68a84"}
+}' -H "Content-Type: application/json"
+```
+
+Note that the broker accepts an `ami_id` parameter ([AMI](http://docs.aws.amazon.com/AWSEC2/latest/UserGuide/AMIs.html) 
+identifier).
+To configure a provisioning request in Service Catalog, we need to declare an `Instance` 
+resource with an AMI identifier declared in the `parameters` field of its spec:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: ami-instance
+  namespace: test-ns
+spec:
+  serviceClassName: aws-ami
+  planName: default
+  parameters:
+    - name: ami_id
+      value: ami-ecb68a84
+```
+
+### Passing sensitive data
+
+To pass a parameter value stored in a `Secret` or a `ConfigMap`, use the `valueFrom` 
+field:
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: username
+      value: admin
+    - name: password
+      valueFrom:
+        secretKeyRef:
+          name: mysecret
+          key: password
+```
+
+### Value types
+
+#### String
+
+By default any parameter value is treated as a string, and passed to the underlying 
+broker "as is", without further validation or processing. If the user needs to pass
+a value of different JSON data type, the field `type` is required to explicitly
+specify data type.
+
+#### JSON object
+
+`type: json` field value declares a value as a JSON object.
+```yaml
+apiVersion: servicecatalog.k8s.io/v1alpha1
+kind: Instance
+metadata:
+  name: qwerty-instance
+  namespace: test-ns
+spec:
+  serviceClassName: qwerty
+  planName: default
+  parameters:
+    - name: txt
+      value: Plain text, passed as a string
+    - name: meta
+      # Explicitly tells that the value is JSON which should be 
+      # validated and passed as a JSON object
```

Comment:
Be clear that "validated" here just means validated that its proper json, right?  

Although, I guess when the OSB API spec includes the schema of the params we could do a more complete syntax validation.